### ISSUE TITLE
Suggestion: New solver Jones-modes

### ISSUE THREAD

**Issue by henedler** [lofar-astron/DP3#344]
The following modes are currently supported by the DP3 solvers: scalar, scalarphase, scalaramplitude, diagonal, diagonalphase, diagonalamplitude, fulljones, tec, tecandphase, rotation and rotation+diagonal .

In a few cases, it would be useful to have additional solve types to reduce free parameters by leveraging known physical effects: 
- **FaradayRotation+diagonal** would help to greatly stabilize the estimate of the polarization misalingment and faraday rotation in the LOFAR LBA calibrator pipeline when also including international stations. The idea is to not solve for one rotation per channel, but to enforce that the rotation follows a RM*lambda^2 frequency dependence. The resulting h5parms would be a diagonal matrix and a faraday000 soltab.
- **FaradayRotation**: Fitting a rotation that follows the RM*lambda^2 frequency dependence. Useful e.g. for DD-calibration in a regime where FR is not negligible (LBA IS, LBA decameter).
- **TEC+clock** would be useful to derive the clock delay of the individual stations in the calibrator pipeline. In general, the function of the form a*nu+b*nu^(-1) shows many comparable local minima, these can be mitigated when using wideband data or constraining the clock delay to be less than one wrap (~LOFAR 2)

Cheers,
Henrik

**Comment by maaijke** [comment]
Hi Henrik,
If it is not too much work it could probably be a good idea to implement these functionalities, but we might need some more discussion on the usability. Clock/TEC fitting on the visibility phases will not lead to optimal solutions, not only because of the local minima issues you mentioned yourself (although it might well work in LOFAR2 if there is a common clock) but also because you are ignoring second order effects like cable reflections and the constant phase offset. Proper Faraday rotation including the 1/frequency^2 effect  is probably easier , so it would certainly be interesting to see  how that goes.

**Comment by henedler** [comment]
I agree that the Clock/TEC issue is probably harder compared to the Faraday constraint, which has priority for us. We currently fit Faraday rotation routinely in solution space using losoto, for the calibrator and the target- this gives good results except for the more distant IS. Fitting FR directly to the visibilities should be more stable since it also allows to take the weights into account.
